crates/sui-source-validation-service/src/lib.rs

```diff
@@ -61,16 +64,108 @@ pub fn parse_config(config_path: impl AsRef<Path>) -> anyhow::Result<Config> {
     toml::from_str(&contents).map_err(anyhow::Error::from)
 }
 
-pub async fn clone_repositories(config: &Config) -> anyhow::Result<()> {
+#[derive(Debug)]
+/// Represents a sequence of git commands to clone a repository and sparsely checkout Move packages within.
+pub struct CloneCommand {
+    /// git args
+    args: Vec<Vec<String>>,
+    /// report repository url in error messages
+    repo_url: String,
+}
+
+impl CloneCommand {
+    pub fn new(p: &Packages, dest: &Path) -> anyhow::Result<CloneCommand> {
+        let repo_url = Url::parse(&p.repository)?;
+        let Some(components) = repo_url.path_segments().map(|c| c.collect::<Vec<_>>()) else {
+	    bail!("Could not discover repository path in url {}", &p.repository)
+	};
+        let Some(repo_name) = components.last() else {
+	    bail!("Could not discover repository name in url {}", &p.repository)
+	};
+        let dest = dest
```

Comment:
`dest` is primarily used to be passed into a command as an argument, in which case we should avoid converting it into a `String` if we can -- keep it as an `OsString`, and same goes for the rest of the arguments.

Reply:
Done! I changed to type so now all command-destined strings are `OsString` instead of `String`